cmd/site-replication.go

```diff
@@ -3919,21 +4194,22 @@ func getSRBucketDeleteOp(isSiteReplicated bool) SRBucketDeleteOp {
 }
 
 func (c *SiteReplicationSys) healBuckets(ctx context.Context, objAPI ObjectLayer) error {
+	info, err := c.siteReplicationStatus(ctx, objAPI, madmin.SRStatusOptions{
+		Users:          true,
+		Policies:       true,
+		Groups:         true,
+		Buckets:        true,
+		ILMExpiryRules: true,
+	})
+	if err != nil {
+		return err
+	}
```

Comment:
fetching site replication status across sites for all buckets in one shot has an issue - depending on number of buckets and other metadata and number of sites in the site replication, this is not a reliable info to use in healing because one/more concurrent updates could happen in the meanwhile.   

Please leave the call in lines 3928-3931 as is and just include your ILMExpiryRules. The ShowDeleted flag is specific to bucket healing to propagate bucket deletes and shouldn't be dropped.

Reply:
Ack